app/actions/backward_compatibility.rb

```diff
@@ -0,0 +1,30 @@
+# All methods is tnis module are DEPRECATED!
+#
+# Do not use them to build anything new.
+#
+# The functions in this module support backward compatibility with parts
+# of the application that we can not just change, e.g. API
+#
+module BackwardCompatibility
```

Comment:
This is a bit over generic imho. How about listing_shape_helper under view_utils instead?